### PR TITLE
Catch error in waitBulkAction. Add bulk.WithRetryOnConflict(3) in multiple places.

### LINKED ISSUE #1894: Bulker swallows errors from elasticsearch when the error is specific to the document
## Overview

Inside of the `BulkIndexerResponseItem` there is an error field that could have possible been set but is ignored in here https://github.com/elastic/fleet-server/blob/main/internal/pkg/bulk/opBulk.go#L77.

## What should be done?

Any code path that reads a `BulkIndexerResponseItem` should check the `Error` field and ensure that an error has not occurred. If an error has occurred then it should return an error instead of continuing on.

Relates to https://github.com/elastic/elastic-agent/issues/760

### PR DESCRIPTION
**Pull request by blakerouse**
## What is the problem this PR solves?

It solves an issue where calling an action through the bulker would result in errors on a single document being ignored. It also adds `bulk.WithRetryOnConflict(3)` in many places that update the `.fleet-agents` index. Being that the `.fleet-agents` index is written to often it is possible that conflicts will occur and elasticsearch should retry on conflicts.

## How does this PR solve the problem?

This changes the `waitBulkAction` to look at the status code and error document of a bulk action response to ensure that an error should not be returned instead of the document.

## How to test this PR locally

Scale test runs are the only place to really test that this solves the issue.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #1894

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - Remove events from agent checkin body. {issue}1774[1774]
 - Improve authc debug logging. {pull}1870[1870]
 - Add error detail to catch-all HTTP error response. {pull}1854[1854]
+- Fix issue where errors where being ignored writting to elasticsearch. {pull}1896[1896]
```
Comment (cmacknz) [inline CHANGELOG.next.asciidoc:13]:
```suggestion
- Fix issue were errors where being ignored written to elasticsearch. {pull}1896[1896]
```

File: internal/pkg/policy/policy_output.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ func (p *Output) prepareElasticsearch(
 			return err
 		}
 
-		if err = bulker.Update(ctx, dl.FleetAgents, agent.Id, body); err != nil {
+		if err = bulker.Update(ctx, dl.FleetAgents, agent.Id, body, bulk.WithRefresh(), bulk.WithRetryOnConflict(3)); err != nil {
```
Comment (cmacknz) [inline internal/pkg/policy/policy_output.go:160]: 1. What does `WithRefresh` do?
2. Is there any place where we don't want these options? Can we make the default so we can't forget the next time someone adds an Update request?

**Review by cmacknz — APPROVED** [review]
LGTM, minus typos in the changelog.

File: internal/pkg/policy/policy_output.go (hunk 2 above)
Comment (blakerouse) [reply internal/pkg/policy/policy_output.go:160]: It is not standard elasticsearch for refresh to be there, and there are places where we do not want refresh. It just happens that in most of these cases we do want refresh.

Comment (blakerouse) [reply internal/pkg/policy/policy_output.go:160]: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html

**Pull request merged**